device/common/pddf/plugins/fanutil.py

```diff
@@ -0,0 +1,214 @@
+            direction = vmap[str(val)]
+
+            print "FAN-%d direction is %s"%(i, direction)
+
+        return 0
+
+    def get_speed(self, idx):
+        # 1 based fan index
+        if idx<1 or idx>self.platform['num_fans']:
+            print "Invalid fan index %d\n"%idx
+            return 0
+
+        attr = "fan" + str(idx) + "_front_rpm"
+        path = pddf_obj.get_path("FAN-CTRL", attr)
+        try:
+            with open(path, 'r') as f:
+                frpm = int(f.read())
+        except IOError:
+            return 0
+
+        return frpm
+
+    def get_speed_rear(self, idx):
+        # 1 based fan index
+        if idx<1 or idx>self.platform['num_fans']:
+            print "Invalid fan index %d\n"%idx
+            return 0
+
+        attr = "fan" + str(idx) + "_rear_rpm"
+        path = pddf_obj.get_path("FAN-CTRL", attr)
+        try:
+            with open(path, 'r') as f:
+                rrpm = int(f.read())
+        except IOError:
+            return 0
+
+        return rrpm
+
+    def get_speeds(self):
+        num_fan = self.get_num_fan();
+        ret = "FAN_INDEX\t\tFRONT_RPM\t\tREAR_RPM\n"
+
+        for i in range(1, num_fan+1):
+            attr1 = "fan" + str(i) + "_front_rpm"
+            attr2 = "fan" + str(i) + "_rear_rpm"
+            path1 = pddf_obj.get_path("FAN-CTRL", attr1)
+            path2 = pddf_obj.get_path("FAN-CTRL", attr2)
+            try:
+                with open(path1, 'r') as f1:
+                    frpm = int(f1.read())
+                with open(path2, 'r') as f2:
+                    rrpm = int(f2.read())
+            except IOError:
+                return False
+
+            ret += "FAN-%d\t\t\t%d\t\t\t%d\n"%(i, frpm, rrpm)
+
+        return ret
+
+    def set_speed(self, val):
+        if val<0 or val>100:
+            print "Error: Invalid speed %d. Please provide a valid speed percentage"%val
+            return False
+        
+        num_fan = self.platform['num_fans']
+        duty_cycle_to_pwm = eval(plugin_data['FAN']['duty_cycle_to_pwm'])
+        pwm = duty_cycle_to_pwm(val)
+        print "New Speed: %d%% - PWM value to be set is %d\n"%(val,pwm)
+
+        status = 0
+        for i in range(1, num_fan+1):
+            attr = "fan" + str(i) + "_pwm"
+            node = pddf_obj.get_path("FAN-CTRL", attr)
+            try:
+                with open(node, 'w') as f:
+                    f.write(str(pwm))
+            except IOError:
+                return False
+
+        #time.sleep(5)
+        #print "Done changing the speed of all the fans ... Reading the speed to crossscheck\n"
+        return True
+
+    def dump_sysfs(self):
+        return pddf_obj.cli_dump_dsysfs('fan')
+
+    def get_change_event(self):
+        """
+        TODO: This function need to be implemented
+        when decide to support monitoring FAN(fand)
+        on this platform.
+        """
+        raise NotImplementedError
+
+
+
+#if __name__== "__main__":
```

Comment:
remove these testing code.

Reply:
Done

Reply:
taken care of in ,
@fk410167
1) GPL license disclaimer inclusion 2) Removal of commented code, as … …
8e8cf2f